test/e2e/main.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+import os
+import sys
+import time
+from difflib import Differ
+from os.path import dirname
+
+import argparse
+import yaml
+import requests
+from requests import Response
+
+try:
+  from yaml import CSafeLoader as Loader
+except ImportError:
+  from yaml import SafeLoader as Loader
+
+def validate(expected_file_name):
+  with open(expected_file_name) as expected_data_file:
+    expected_data = os.linesep.join(expected_data_file.readlines())
+    response = requests.post(url='http://0.0.0.0:12800/dataValidate', data=expected_data)
+
+    if response.status_code != 200:
+      res = requests.get('http://0.0.0.0:12800/receiveData')
+      actual_data = yaml.dump(yaml.load(res.content, Loader=Loader))
+
+      differ = Differ()
+      diff_list = list(differ.compare(
+        actual_data.splitlines(keepends=True),
+        yaml.dump(yaml.load(expected_data, Loader=Loader)).splitlines(keepends=True)
+      ))
+
+      print('diff list: ')
+      sys.stdout.writelines(diff_list)
+
+    assert response.status_code == 200
+    return response
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser()
+  parser.add_argument('expected_file', help='File name which includes expected reported value')
+  args = parser.parse_args()
+
+  requests.get('http://0.0.0.0:8081/ping')
```

Comment:
Same here, better have some conditions that the `docker-compose`d services are ready/healthy

Reply:
`main.py` will be execute after all of containers started. https://github.com/SkyAPM/cpp2sky/pull/18/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R30-R39 So I think that we don't have to aware it. no?

Reply:
The problem is that all containers being started doesn't mean they are ready, just because these steps (circled below) cost some time and those containers are initialized quickly, so that before `main.py` is really executed, all the containers are ready, but it's kind of potential instability if the containers get slower to initialized, anyway with docker-compose > 3, I haven't come out with a good idea, it's ok to just keep it as is for now until we have to

![image](https://user-images.githubusercontent.com/15965696/100880226-73cf7e80-34e7-11eb-841f-ae41b58f7288.png)

Reply:
resolved with application level retry. thanks!